extra_scripts/create_inaturalist2018_data_files.py

```diff
@@ -0,0 +1,120 @@
+        "https://ml-inat-competition-datasets.s3.amazonaws.com/2018/val2018.json.tar.gz"
+    )
+    download_and_extract_archive(url=TRAINING_ANNOTATIONS_URL, download_root=input)
+    download_and_extract_archive(url=VAL_ANNOTIATIONS_URL, download_root=input)
+
+    logger.info("========Downloading Images. This may take awhile! ========")
+    IMAGES_URL = "https://ml-inat-competition-datasets.s3.amazonaws.com/2018/train_val2018.tar.gz"
+    download_and_extract_archive(url=IMAGES_URL, download_root=input)
+
+
+def get_images_labels_info(input_file_name: str, input: str):
+    """
+    Process the iNaturalist2018 image paths and labels for the training or validation data
+    This assumes the input contains the train2018.json and val2018.json files with relative paths to
+    training and validation images, respectively, as well as a sub-directory with all the raw images
+    """
+    data_source_path = input + input_file_name
+    with (open(data_source_path)) as ds_path:
+        data = json.load(ds_path)
+
+    images, labels = [], []
+
+    for info in data["images"]:
+        image = info["file_name"]
+        path_elements = image.split("/")
+        full_path = input + "/train_val2018/" + "/".join(path_elements[1:])
+        label = int(path_elements[2])
+        images.append(full_path)
+        labels.append(label)
+    return np.array(images), np.array(labels)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Create the iNaturalist2018 data information file."
+    )
+    parser.add_argument(
+        "-i",
+        "--input",
+        type=str,
+        help="Path to the parent directory of the iNaturalist2018 data set",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=str,
+        help="Folder where output npy files (that wil be called train|val_images.npy and train|val_labels.npy) are generated",
+    )
+    parser.add_argument(
+        "-d",
+        "--download",
+        action="store_const",
+        const=True,
+        default=False,
+        help="To download the original dataset and decompress it in the input folder",
+    )
+    args = parser.parse_args()
+
+    # Download dataset to input path
+    if args.download:
+        download_dataset(args.input)
+
+    # Process training and validation datasets into numpy arrays
+    logger.info("========Preparing train data files========")
+    train_images, train_labels = get_images_labels_info("/train2018.json", args.input)
+    logger.info("========Preparing val data files========")
+    val_images, val_labels = get_images_labels_info("/val2018.json", args.input)
+
+    # Save as numpy files to output path
+    logger.info("=================Saving train data files=======================")
+    train_label_file_name = args.output + "/train_labels.npy"
+    train_image_file_name = args.output + "/train_images.npy"
+    np.save(train_image_file_name, train_images)
+    np.save(train_label_file_name, train_labels)
```

Comment:
nit: we should use PathManager. Checkout https://github.com/facebookresearch/vissl/blob/master/extra_scripts/create_imagenet_data_files.py#L16 . this will ensure that several backends (beyond disk) will be supported to save the data :)